src/Umbraco.Web.UI.Client/src/views/common/overlays/treepicker/treepicker.controller.js

```diff
@@ -186,9 +186,18 @@ angular.module("umbraco").controller("Umbraco.Overlays.TreePickerController",
             var nodeHasPath = typeof node !== "undefined" && typeof node.path !== "undefined";
             var startNodeNotDefined = typeof dialogOptions.startNodeId === "undefined" || dialogOptions.startNodeId === "" || dialogOptions.startNodeId === "-1";
             if (startNodeNotDefined && nodeHasPath) {
+                // since the node expanded event isn't fired while syncing the tree, we need to do some post filtering

```

Comment:
Alternatively, in the treeSyncedHandler, you could walk back up the tree to the root and then back down using the passed in node in `args` instead of relying on saving the first expanded set of nodes.